poetry.lock

```diff
@@ -44,8 +44,8 @@ test = ["anyio[trio] (>=3.2.1,<4.0.0)", "black (==23.1.0)", "coverage[toml] (>=6
 [package.source]
 type = "git"
 url = "ssh://git@github.com/tiangolo/fastapi.git"
-reference = "0.95.0"
-resolved_reference = "d666ccb62216e45ca78643b52c235ba0d2c53986"
+reference = "HEAD"
```

Comment:
Why does this change from a version to non version?

Reply:
I supposed this use case was easier to implement. Still given it's not the case, would you be ok to 
switch to our ideal use case (https://github.com/yetanotherion/poetry-git-dependency-repro/pull/4)